app/src/main/java/com/arygm/quickfix/ui/uiMode/appContentUI/userModeUI/search/SearchOnBoarding.kt

```diff
@@ -1,212 +1,394 @@
+          filterState.phoneLocation = userLoc
+        } else {
+          Toast.makeText(context, "Unable to fetch location", Toast.LENGTH_SHORT).show()
+        }
+      }
+    } else {
+      Toast.makeText(context, "Enable Location In Settings", Toast.LENGTH_SHORT).show()
+    }
+    uid = loadUserId(preferencesViewModel)
+    userProfileViewModel.fetchUserProfile(uid) { profile -> userProfile = profile as UserProfile }
+  }
+  fun updateFilteredProfiles() {
+    filteredWorkerProfiles = filterState.reapplyFilters(workerProfiles, searchViewModel)
+  }
+
+  // Build filter buttons
+  val listOfButtons =
+      filterState.getFilterButtons(
+          workerProfiles = workerProfiles,
+          filteredProfiles = filteredWorkerProfiles,
+          searchViewModel = searchViewModel,
+          onProfilesUpdated = { updated -> filteredWorkerProfiles = updated },
+          onShowAvailabilityBottomSheet = {
+            setUiState(uiState.copy(showAvailabilityBottomSheet = true))
+          },
+          onShowServicesBottomSheet = { setUiState(uiState.copy(showServicesBottomSheet = true)) },
+          onShowPriceRangeBottomSheet = {
+            setUiState(uiState.copy(showPriceRangeBottomSheet = true))
+          },
+          onShowLocationBottomSheet = { setUiState(uiState.copy(showLocationBottomSheet = true)) })
+
+  val profileImagesMap by remember { mutableStateOf(mutableMapOf<String, Bitmap?>()) }
+  val bannerImagesMap by remember { mutableStateOf(mutableMapOf<String, Bitmap?>()) }
+  var loading by remember { mutableStateOf(true) }
+  // Tracks if data is loading
+  LaunchedEffect(workerProfiles) {
+    if (workerProfiles.isNotEmpty()) {
+      workerProfiles.forEach { profile ->
+        // Fetch profile images
+        workerViewModel.fetchProfileImageAsBitmap(
+            profile.uid,
+            onSuccess = { bitmap ->
+              profileImagesMap[profile.uid] = bitmap
+              checkIfLoadingComplete(workerProfiles, profileImagesMap, bannerImagesMap) {
+                loading = false
+              }
+            },
+            onFailure = { Log.e("ProfileResults", "Failed to fetch profile image") })
 
+        // Fetch banner images
+        workerViewModel.fetchBannerImageAsBitmap(
+            profile.uid,
+            onSuccess = { bitmap ->
+              bannerImagesMap[profile.uid] = bitmap
+              checkIfLoadingComplete(workerProfiles, profileImagesMap, bannerImagesMap) {
+                loading = false
+              }
+            },
+            onFailure = { Log.e("ProfileResults", "Failed to fetch banner image") })
+      }
+    } else {
+      loading = false // No profiles to load
+    }
+  }
   BoxWithConstraints {
     val widthRatio = maxWidth.value / 411f
     val heightRatio = maxHeight.value / 860f
     val sizeRatio = minOf(widthRatio, heightRatio)
-    val screenHeight = maxHeight.value
-    val screenWidth = maxWidth.value
+    val screenHeight = maxHeight
+    val screenWidth = maxWidth
```

Comment:
These changes remove a fix to a buggy representation of the search bar on smaller phones. Drop these changes please.¹